features/kafka/producer/src/test/java/org/opennms/features/kafka/producer/KafkaEventForwarderIT.java

```diff
@@ -0,0 +1,261 @@
+        "classpath:/META-INF/opennms/applicationContext-minimal-conf.xml",
+        "classpath:/META-INF/opennms/applicationContext-mockDao.xml",
+        "classpath:/META-INF/opennms/applicationContext-daemon.xml",
+        "classpath:/META-INF/opennms/mockEventIpcManager.xml",
+        "classpath:/META-INF/opennms/applicationContext-alarmd.xml",
+        "classpath:/applicationContext-test-kafka-producer.xml" })
+@JUnitConfigurationEnvironment
+@JUnitTemporaryDatabase(dirtiesContext = false, tempDbClass = MockDatabase.class, reuseDatabase = false)
+public class KafkaEventForwarderIT implements TemporaryDatabaseAware<MockDatabase> {
+
+    private static final String KAFKA_PRODUCER_CLIENT_PID = "org.opennms.features.kafka.producer.client";
+    private static final int NODE_ID_ONE = 1;
+
+    @Rule
+    public JUnitKafkaServer kafkaServer = new JUnitKafkaServer();
+
+    @Autowired
+    private MockEventIpcManager m_eventdIpcMgr;
+
+    @Autowired
+    private ProtobufMapper protobufMapper;
+
+    @Autowired
+    private NodeCache nodeCache;
+
+    @Autowired
+    private AlarmLifecycleListenerManager alarmLifecycleListenerManager;
+
+    @Autowired
+    private MonitoringLocationDao m_locationDao;
+
+    @Autowired
+    private NodeDao m_nodeDao;
+
+    private OpennmsKafkaProducer kafkaProducer;
+
+    private KafkaMessageConsumerRunner kafkaConsumer;
+
+    private MockDatabase m_database;
+
+    @Before
+    public void setup() throws IOException, InterruptedException {
+
+        m_eventdIpcMgr.setEventWriter(m_database);
+        OnmsMonitoringLocation location = new OnmsMonitoringLocation();
+        location.setLocationName("Default");
+        m_locationDao.save(location);
+        final OnmsNode node = new OnmsNode(location, "node1");
+        node.setId(NODE_ID_ONE);
+        m_nodeDao.save(node);
+
+        Hashtable<String, Object> producerConfig = new Hashtable<String, Object>();
+        producerConfig.put("group.id", "OpenNMS");
+        producerConfig.put("bootstrap.servers", kafkaServer.getKafkaConnectString());
+        ConfigurationAdmin configAdmin = mock(ConfigurationAdmin.class, RETURNS_DEEP_STUBS);
+        when(configAdmin.getConfiguration(KAFKA_PRODUCER_CLIENT_PID).getProperties()).thenReturn(producerConfig);
+
+        ExecutorService executor = Executors.newSingleThreadExecutor();
+        kafkaConsumer = new KafkaMessageConsumerRunner(kafkaServer.getKafkaConnectString());
+        executor.execute(kafkaConsumer);
+
+        kafkaProducer = new OpennmsKafkaProducer(protobufMapper, nodeCache, configAdmin, m_eventdIpcMgr,
+                alarmLifecycleListenerManager);
+
+        kafkaProducer.setEventTopic("events");
+        kafkaProducer.setEventFilter("getUei().equals(\"uei.opennms.org/internal/discovery/newSuspect\")");
+        kafkaProducer.setNodeTopic("nodes");
+        kafkaProducer.init();
+        // 2 second delay for kafka producer to initialize
+        Thread.sleep(2000);
```

Comment:
What if it takes longer to initialize? Would it be possible to use something like `await().atMost(1, MINUTES).until(() -> kafkaProducer.isReady())`?